sdk/communication/AzureCommunicationChat/Source/Signaling/TrouterEventUtil.swift

```diff
@@ -0,0 +1,131 @@
+public enum TrouterEventUtil {
+    /// Parses out the id/phone number portion of an MRI.
+    /// - Parameters:
+    ///   - mri: The original MRI.
+    ///   - prefix: The MRI prefix.
+    /// - Returns: The part of the MRI after the prefix that corresponds to the id or phone number of a user.
+    private static func parse(mri: String, prefix: String) -> String {
+        let index = mri.index(mri.startIndex, offsetBy: prefix.count)
+        return String(mri.suffix(from: index))
+    }
+
+    /// Constructs a CommunicationIdentifier from an MRI.
+    /// - Parameter mri: The MRI.
+    /// - Returns: The CommunicationIdentifier.
+    public static func getIdentifier(from mri: String) -> CommunicationIdentifier {
+        let publicTeamsUserPrefix = "8:orgid:"
+        let dodTeamsUserPrefix = "8:dod:"
+        let gcchTeamsUserPrefix = "8:gcch:"
+        let teamsVisitorUserPrefix = "8:teamsvisitor:"
+        let phoneNumberPrefix = "4:"
+        let acsUserPrefix = "8:acs:"
+        let spoolUserPrefix = "8:spool:"
+
+        if mri.starts(with: publicTeamsUserPrefix) {
+            return MicrosoftTeamsUserIdentifier(
+                userId: parse(mri: mri, prefix: publicTeamsUserPrefix),
+                isAnonymous: false,
+                rawId: mri,
+                cloudEnvironment: CommunicationCloudEnvironment.Public
+            )
+        } else if mri.starts(with: dodTeamsUserPrefix) {
+            return MicrosoftTeamsUserIdentifier(
+                userId: parse(mri: mri, prefix: dodTeamsUserPrefix),
+                isAnonymous: false,
+                rawId: mri,
+                cloudEnvironment: CommunicationCloudEnvironment.Dod
+            )
+        } else if mri.starts(with: gcchTeamsUserPrefix) {
+            return MicrosoftTeamsUserIdentifier(
+                userId: parse(mri: mri, prefix: gcchTeamsUserPrefix),
+                isAnonymous: false,
+                rawId: mri,
+                cloudEnvironment: CommunicationCloudEnvironment.Gcch
+            )
+        } else if mri.starts(with: teamsVisitorUserPrefix) {
+            return MicrosoftTeamsUserIdentifier(
+                userId: parse(mri: mri, prefix: teamsVisitorUserPrefix),
+                isAnonymous: true
+            )
+        } else if mri.starts(with: phoneNumberPrefix) {
+            return PhoneNumberIdentifier(
+                phoneNumber: parse(mri: mri, prefix: phoneNumberPrefix),
+                rawId: mri
+            )
+        } else if mri.starts(with: acsUserPrefix) || mri.starts(with: spoolUserPrefix) {
+            return CommunicationUserIdentifier(mri)
+        } else {
+            return UnknownIdentifier(mri)
+        }
+    }
+
+    public static func toISO8601Date(unixTime: Int? = 0) -> String {
+        let unixTimeInMilliSeconds = Double(unixTime ?? 0) / 1000
+        let date = Date(timeIntervalSince1970: TimeInterval(unixTimeInMilliSeconds))
+        let iso8601DateFormatter = ISO8601DateFormatter()
```

Comment:
Should be `Iso8601DateFormatter`

Reply:
the naming should start with lower case character, or lint complains

Reply:
I mean the class name, not the variable name.